pydantic/__init__.py

```diff
@@ -159,5 +162,18 @@
     'VERSION',
 ]
 
+# A mapping of {<member name>: <module name>} defining dynamic imports
+__all_dynamic__ = {'RootModel': '.root_model'}
+if typing.TYPE_CHECKING:
+    from .root_model import *
+
+__getattr_migration__ = getattr_migration(__name__)
+
+
+def __getattr__(attr_name: str) -> object:
+    if attr_name in __all_dynamic__:
```

Comment:
```suggestion
    attr = _dynamic_imports.get(attr_name
    if attr is None:
        return _getattr_migration(attr_name)
    else:
        ...
```
Better to look up `_dynamic_imports` once not twice.